src/main/java/net/minecraftforge/fml/common/network/FMLOutboundHandler.java

```diff
@@ -43,9 +44,10 @@
 import javax.annotation.Nullable;
 
 public class FMLOutboundHandler extends ChannelOutboundHandlerAdapter {
-    public static final AttributeKey<OutboundTarget> FML_MESSAGETARGET = AttributeKey.valueOf("fml:outboundTarget");
+    public static final AttributeKey<IOutboundTarget> FML_MESSAGETARGET = AttributeKey.valueOf("fml:outboundTarget");
```

Comment:
It seems to not be binary incompatible, but may cause a runtime exception in the unlikely event where it is set to a non-OutboundTarget.

Looking at line 300 in this file with a bytecode inspector, you will find it will CHECKCAST the value of FML_MESSAGETARGET to the type it is compiled with.

```
    LINENUMBER 300 L7
    ALOAD 1
    INVOKEINTERFACE io/netty/channel/ChannelHandlerContext.channel ()Lio/netty/channel/Channel;
    GETSTATIC net/minecraftforge/fml/common/network/FMLOutboundHandler.FML_MESSAGETARGET : Lio/netty/util/AttributeKey;
    INVOKEINTERFACE io/netty/channel/Channel.attr (Lio/netty/util/AttributeKey;)Lio/netty/util/Attribute;
    INVOKEINTERFACE io/netty/util/Attribute.get ()Ljava/lang/Object;
    CHECKCAST net/minecraftforge/fml/common/network/IOutboundTarget
    ASTORE 5
   L10
```

It will, however, cause compilation errors in mods that use FML_MESSAGETARGET.